### ISSUE TITLE
[3.6][Bug][Has Fix] Couple of issues with table field

### ISSUE THREAD

**Issue by danielbidala** [Laravel-Backpack/CRUD#1652]
Hi guys, I have these problems with table field. I'm the only one with issues below?

**Case1:** With casting to array.

1. Issue: If you create a new record and add two row to table field and the second row is empty it will save the second (empty) row to json too. If you edit this record you get a json in your hidden input like this:

`[{"db":"value","type":"value","manufacturer":"value"},[]]`
instead of this
`[{"db":"value","type":"value","manufacturer":"value"},{}]`

Now if you add values to second row it won't be saved!

2. Issue: If you add the table column to your `show()` method and if you have empty field(s) in your table field row you get the following error when you open show view:

Undefined index: **type** (View: /home/brightstaff/laravel_backpack/vendor/backpack/crud/src/resources/views/columns/table.blade.php) (View: /home/brightstaff/laravel_backpack/vendor/backpack/crud/src/resources/views/columns/table.blade.php)

Where **type** is the first field that you leaved empty in the row.

**Case1:** With casting to object or without casting

1. Issue: The empty row problem is gone away when you use casting to object or you don't setup casting.
2. Issue: If you add the table column to your `show()` method get the following error when you open show view:

Cannot use object of type stdClass as array (View: /home/brightstaff/laravel_backpack/vendor/backpack/crud/src/resources/views/columns/table.blade.php) (View: /home/brightstaff/laravel_backpack/vendor/backpack/crud/src/resources/views/columns/table.blade.php)

I use:
PHP 7.1
Laravel 5.7
BackPack 3.4

**Comment by danielbidala** [comment]
I go with casting as object and override the `table.blade.php` column type with an `if isset` to solve this temporary:

```
@foreach($columns as $tableColumnKey => $tableColumnLabel)
		<td>
			<?php if (isset($tableRow->$tableColumnKey)):?>
			{{ $tableRow->{$tableColumnKey} ?? $tableRow[$tableColumnKey] }}
			<?php endif; ?>
		</td>
@endforeach
```

**Comment by tabacitu** [comment]
Thanks for the bug report @danielbidala ! Should definitely be fixed - inconsistent UX sucks. **Do you want to create a small PR with this fix**, so you rightfully show up as a Backpack contributor? Happy to test & merge it if you do. You should be able to easily do this one from the Github interface.

Let me know.

Cheers!

PS. Let's leave this open until it's solved.

**Comment by tabacitu** [comment]
Hi @danielbidala ,

You're totally right - thank you. I've just merged the fixes into ```3.6.28```, it'll be available with a ```composer update``` later today,  after I merge a few more non-breaking changes and tag it.

Thanks again! And sorry for taking so long to fix this.
Cheers!
